src/node.h

```diff
@@ -362,6 +362,13 @@ NODE_EXTERN v8::Local<v8::Context> NewContext(
     v8::Local<v8::ObjectTemplate> object_template =
         v8::Local<v8::ObjectTemplate>());
 
+// Create a new context for an existing environment.
+// This add several fields to make inspector work properly.
```

Comment:
I would disagree, since there are three major problems in what you have described:
- The API through which ``v8_inspector::V8Inspector`` is accessed must be aware both in compile-time and in runtime whether an inspector agent actually exists (currently ``node::Environment`` does that);
- ``v8_inspector::V8Inspector::ContextCreated()`` is intended to use from the context-creating functions. It is not a notification and it will happily re-assign the ``debug_context_id`` leading to undefined behavior for any outstanding requests from the frontend with the old ``debug_context_id``.
- In the V8 API there are per-isolate properties like ``v8::Isolate::SetPrepareStackTraceCallback()``. If the environment does not handle that there should be another class in the hierarchy: ``node::MultiIsolatePlatform -> (per-isolate object) -> node::Environment``. This would require major untangling of the ``node::Environment`` and it is not related to the main goal here: creating a new context.

I think that hiding the context creation process steps from the API is way more stable than exposing the inspector agent to the embedder. Currently ``node::Environment`` perform some per-isolate processes like ``v8::Isolate::SetPrepareStackTraceCallback``. In the V8 engine ``v8::Context`` objects are part of ``v8::Isolate``, and there are certain API that are per-isolate. Therefore, if ``node::Environment`` does not handle the per-isolate API, there must be another class that handles it (and ``node::Environment`` would probably contains shared pointer towards it). And since the inspector agent would always be hidden within an instance of a node's class (see the first two points above), the best approach for the context creation process would be to have ``node::NewContext`` that accepts the node's per-isolate object and perform node's specific initialization (like reporting the context to its inspector agent).

For the current minor version, ``node::Environment`` is still per-isolate, and the exposed function signature allow proper initialization with provided embedder's ``v8::ObjectTemplate`` while reporting to the inspector agent is not exposed and how it is done can be changed in the future versions.

The only problem is the ``initialization`` argument. To become future-proof compatible, it would be better if the third parameter is some white-box (publicly exposed) structure, that can be expanded in the future versions.